framework/wcf/Extensibility/Transport/AzureQueueStorageClient/AzureQueueStorageClient/InputQueue.cs

```diff
@@ -0,0 +1,1086 @@
+﻿//----------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//----------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Microsoft.Samples.AzureQueueStorage
+{
+
+
+    // ItemDequeuedCallback is called as an item is dequeued from the InputQueue.  The 
+    // InputQueue lock is not held during the callback.  However, the user code is
+    // not notified of the item being available until the callback returns.  If you
+    // are not sure if the callback blocks for a long time, then first call 
+    // IOThreadScheduler.ScheduleCallback to get to a "safe" thread.
+    delegate void ItemDequeuedCallback();
+
+    /// <summary>
+    /// Handles asynchronous interactions between producers and consumers. 
+    /// Producers can dispatch available data to the input queue, 
+    /// where it is dispatched to a waiting consumer or stored until a
+    /// consumer becomes available. Consumers can synchronously or asynchronously
+    /// request data from the queue, which is returned when data becomes
+    /// available.
+    /// </summary>
+    /// <typeparam name="T">The concrete type of the consumer objects that are waiting for data.</typeparam>
+    class InputQueue<T> : IDisposable where T : class
+    {
```

Comment:
You should be able to delete this class as it's used for the server side listener in WCF.

Reply:
removed